packages/client/src/api/wargames_api/index.ts

```diff
@@ -509,6 +511,34 @@ export const deleteForce = (dbName: string, forceName: string): Promise<Wargame>
     forces.splice(forceIndex, 1)
     updatedData.forces.forces = forces
     updatedData.channels.complete = calcComplete(forces)
+    updatedData.channels.channels.forEach((v) => (v.participants as CoreParticipant[] | ForceData[]) = forces)
+    if (forces.length === 0) {
+      updatedData.channels = { 
+      name: 'Channels',
+      channels: [],
+      selectedChannel: '',
+      complete: false,
+      dirty: false
+      }
+    }
+    return updateWargame({ ...res, data: updatedData }, dbName)
+  })
+}
+
+export const deleteRolesParticipations = (dbName: string, roles: Role[]): Promise<Wargame> => {
+  return getLatestWargameRevision(dbName).then((res) => {
+    const newDoc: Wargame = deepCopy(res)
+    const updatedData = newDoc.data
```

Comment:
@lilitkarapetyan - there is a mistake in the below code.  It is setting `roles` for every force - they all get given the new set of roles.  So, if I delete a `blue` role, the new list of roles gets stored for `white` and `red`, too.

Reply:
Steps to reproduce:
- login as admin
- open `Small wargame`
- open `forces`
- open `White` then `roles`
- view 4 roles
- now open `Blue-1` roles
- delete `CT5-B5`
- click `Save Force`
- view `White` then `roles
- now see there is just the `CO` role for `White` (there should be 4 of them)
- also see it's the same single role for `Red`, too.

IMHO, `deleteRolesParticipations` should receive a `force-id` and `role-id` - so it can remove the correct `role` from the correct `force`.